components/hts221/hts221.h

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2021 saasaa <mail@saasaa.xyz>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+/**
+ * @file hts221.h
+ * @brief HTS221 driver
+ * @author saasaa
+ * @defgroup hts221 hts221
+ * @{
+ *
+ * ESP-IDF driver for HTS221 temperature and humidity sensor
+ *
+ * Datasheet: http://www.st.com/resource/en/datasheet/hts221.pdf
+ * Technical note on interpreting humidity and temperature readings in the
+ * HTS221 digital humidity sensor:
+ * https://www.st.com/resource/en/technical_note/tn1218-interpreting-humidity-and-temperature-readings-in-the-hts221-digital-humidity-sensor-stmicroelectronics.pdf
+ * Copyright (c) 2021 saasaa <mail@saasaa.xyz>
+ *
+ * ISC Licensed as described in the file LICENSE
+ *
+ */
+
+#ifndef __HTS221_H__
+#define __HTS221_H__
+
+#include <esp_err.h>
+#include <i2cdev.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define HTS221_I2C_ADDRESS 0x5F
+
+#define HTS221_REG_WHOAMI 0x0F   // R   Device identification register
```

Comment:
Are these register numbers somehow used outside the library? It seems to me that this is purely internal information and has no place in the header file.